src/student_success_tool/analysis/pdp/targets/failure_to_graduate_within_x_years.py

```diff
@@ -0,0 +1,192 @@
+    n: int,
+) -> pd.DataFrame:
+    """
+    Make a labeled dataset for modeling, where each row consists of features
+    from eligible students' first term (within their cohort)
+    matched to computed target variables.
+
+    See Also:
+        - :func:`select_eligible_students()`
+        - :func:`compute_target_variable()`
+        - :func:`shared.get_first_student_terms_within_cohort()`
+    """
+    df_eligible_students = select_eligible_students(
+        df,
+        student_criteria=student_criteria,
+        student_id_cols=student_id_cols,
+        intensity_time_lefts=intensity_time_lefts,
+        max_term_rank=max_term_rank,
+        num_terms_in_year=num_terms_in_year,
+        exclude_pre_cohort_terms=exclude_pre_cohort_terms,
+        term_is_pre_cohort_col=term_is_pre_cohort_col,
+        n=n,
+        enrollment_intensity_col=enrollment_intensity_col,
+        term_rank_col=term_rank_col,
+    )
+    df_eligible_student_terms = pd.merge(
+        df, df_eligible_students, on=student_id_cols, how="inner"
+    )
+    if exclude_pre_cohort_terms is True:
+        df_features = shared.get_nth_student_terms_within_cohort(
+            df_eligible_student_terms,
+            student_id_cols=student_id_cols,
+            term_is_pre_cohort_col=term_is_pre_cohort_col,
+            sort_cols=term_rank_col,
+            n=n,
+            include_cols=None,
+        )
+    else:
+        df_features = shared.get_nth_student_terms(
+            df_eligible_student_terms,
+            student_id_cols=student_id_cols,
+            sort_cols=term_rank_col,
+            n=n,
+            include_cols=None,
+        )
+    df_targets = compute_target_variable(
+        df_eligible_student_terms,
+        intensity_time_lefts=intensity_time_lefts,
+        student_id_cols=student_id_cols,
+        enrollment_intensity_col=enrollment_intensity_col,
+        years_to_degree_col=years_to_degree_col,
+        num_terms_in_year=num_terms_in_year,
+    )
+    df_labeled = pd.merge(df_features, df_targets, on=student_id_cols, how="inner")
+    return df_labeled
+
+
+def compute_target_variable(    
+    df: pd.DataFrame,    
+    *,    
+    student_id_cols: str | list[str] = "student_guid",    
+    years_to_degree_col: str = "first_year_bach",  
+    intensity_time_lefts: list[tuple[str, float, t.Literal["year", "term"]]],   
+    intensity_col: str = "intensity_type",  
+    enrollment_intensity_col: str = "enrollment_intensity_first_term",
+    num_terms_in_year: int = 4,
+) -> pd.Series:   
+    """  
+    Args:  
+        df  
+        student_id_cols  
+        years_to_degree_col  
+        intensity_time_lefts  
+        intensity_col  
+    """   
+    student_id_cols = utils.to_list(student_id_cols)  
+    intensity_num_terms = {intensity: time if unit == "year" else time / num_terms_in_year  
+                           for intensity, time, unit in intensity_time_lefts  
+                           }   
+    return (    
```

Comment:
@bdewilde and here too please! I tried what you had wrote, it was a little buggy and wasn't labeling correctly. I think that may have been for other reasons that I've now fixed so I'll test it again in a bit - but this current code is tested and works!

Reply:
I'm reasonably confident in what I wrote, since it was adapted from existing code that definitely does work -- can you be more specific about what was "buggy"? It's hard to respond effectively to prompts like "let me know what you think", I need a little more specificity in what you're wondering about.

Reply:
It needed a few tweaks but it's good now!